src/feature_filter/feature_filter.test.ts

```diff
@@ -328,10 +328,88 @@ function legacyFilterTests(createFilterExpr) {
-        expect(f({zoom: 0}, {type: 1})).toBe(false);
-        expect(f({zoom: 0}, {type: 2})).toBe(true);
+    test('==, $type, Point', () => {
+        const fPoint = createFilterExpr(['==', '$type', 'Point']).filter;
+        expect(fPoint({zoom: 0}, {type: 0})).toBe(false);       // Unknown geometry-type
+        expect(fPoint({zoom: 0}, {type: 1, geometry: [[{x: 0, y: 0}]]})).toBe(true);    // Point geometry-type
+        expect(fPoint({zoom: 0}, {type: 1, geometry: [[{x: 0, y: 0}], [{x: 0, y: 0}]]})).toBe(true);    // MultiPoint geometry-type
+        expect(fPoint({zoom: 0}, {type: 2, geometry: [[{x: 0, y: 0}, {x: 1, y: 1}]]})).toBe(false);     // Linestring geometry-type
+        expect(fPoint({zoom: 0}, {type: 2, geometry: [[{x: 0, y: 0}, {x: 1, y: 1}], [{x: 2, y: 0}, {x: 1, y: 0}]]})).toBe(false); // MultiLineString geometry-type
+        expect(fPoint({zoom: 0}, {type: 3, geometry: [[{x: 0, y: 0}, {x: 3, y: 0}, {x: 3, y: 3}, {x: 0, y: 3}, {x: 0, y: 0}],
+            [{x: 0, y: 0}, {x: 0, y: 2}, {x: 2, y: 2}, {x: 2, y: 0}, {x: 0, y: 0}]]})).toBe(false);     // Polygon geometry-type
+        expect(fPoint({zoom: 0}, {type: 3, geometry: [[{x: 0, y: 0}, {x: 3, y: 0}, {x: 3, y: 3}, {x: 0, y: 3}, {x: 0, y: 0}],
+            [{x: 0, y: 0}, {x: 0, y: 2}, {x: 2, y: 2}, {x: 2, y: 0}, {x: 0, y: 0}],
+            [{x: 0, y: 0}, {x: 1, y: 0}, {x: 1, y: 1}, {x: 0, y: 1}, {x: 0, y: 0}]]})).toBe(false); // MultiPolygon geometry-type
+        expect(fPoint({zoom: 0}, {type: 'Unknown'})).toBe(false);
+        expect(fPoint({zoom: 0}, {type: 'Point'})).toBe(true);
+        expect(fPoint({zoom: 0}, {type: 'MultiPoint'})).toBe(true);
+        expect(fPoint({zoom: 0}, {type: 'LineString'})).toBe(false);
+        expect(fPoint({zoom: 0}, {type: 'MultiLineString'})).toBe(false);
+        expect(fPoint({zoom: 0}, {type: 'Polygon'})).toBe(false);
+        expect(fPoint({zoom: 0}, {type: 'MultiPolygon'})).toBe(false);
+    });
+
+    test('==, $type, LineString', () => {
+        const fLineString = createFilterExpr(['==', '$type', 'LineString']).filter;
+        expect(fLineString({zoom: 0}, {type: 0})).toBe(false);  // Unknown geometry-type
+        expect(fLineString({zoom: 0}, {type: 1, geometry: [[{x: 0, y: 0}]]})).toBe(false);      // Point geometry-type
+        expect(fLineString({zoom: 0}, {type: 1, geometry: [[{x: 0, y: 0}], [{x: 0, y: 0}]]})).toBe(false);      // MultiPoint geometry-type
+        expect(fLineString({zoom: 0}, {type: 2, geometry: [[{x: 0, y: 0}, {x: 1, y: 1}]]})).toBe(true); // Linestring geometry-type
+        expect(fLineString({zoom: 0}, {type: 2, geometry: [[{x: 0, y: 0}, {x: 1, y: 1}], [{x: 2, y: 0}, {x: 1, y: 0}]]})).toBe(true); // MultiLineString geometry-type
+        expect(fLineString({zoom: 0}, {type: 3, geometry: [[{x: 0, y: 0}, {x: 3, y: 0}, {x: 3, y: 3}, {x: 0, y: 3}, {x: 0, y: 0}],
+            [{x: 0, y: 0}, {x: 0, y: 2}, {x: 2, y: 2}, {x: 2, y: 0}, {x: 0, y: 0}]]})).toBe(false);     // Polygon geometry-type
```

Comment:
The comments here are also probably an indication of a further need to split tests I guess. 
Better to have a string for test name instead of a comment if you'd ask me.